db/helpers/generatorUtils.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * db/helpers/generatorUtils.js
+ *
+ * DB utils for the generator model.
+ */
+const crypto = require('crypto');
+const dbConstants = require('../constants');
+
+/**
+ * Encrypt the given data using passed in secretKey and algorithm
+ * @param  {String} data -  Data to be encrypted
+ * @param  {String} secretKey - Secret key for encryption
+ * @param  {String} algorithm - Name of the encryption algorithm
+ * @returns {String} - encrypted data
+ */
+function encrypt(data, secretKey, algorithm) {
+  const cipher = crypto.createCipher(algorithm, secretKey);
+  let crypted = cipher.update(data, 'utf-8', 'hex');
+  crypted += cipher.final('hex');
+  return crypted;
+} // encrypt
+
+/**
+ * Decrypt encrypted data using passed in secretKey and algorithm
+ * @param  {String} data -  Data to be decrypted
+ * @param  {String} secretKey - Secret key that was used from encryption
+ * @param  {String} algorithm - Name of the encryption algorithm
+ * @returns {String} - encrypted data
+ */
+function decrypt(data, secretKey, algorithm) {
+  const decipher = crypto.createDecipher(algorithm, secretKey);
+  let decrypted = decipher.update(data, 'hex', 'utf-8');
+  decrypted += decipher.final('utf-8');
+  return decrypted;
+} // decrypt
+
+/**
+ * This function fetches the sample generator secret key and the sample
+ * generator encryption algorithm from the glocalConfig model
+ * and returns an object containing the secretKey and algorithm
+ * @param {Model} globalConfigModel - The sequelize Global Config Model
+ * @returns {Object} - containing the secret key and algorithm
+ */
+function getSGEncryptionKeyAndAlgorithm(globalConfigModel) {
+  const sgObj = {};
+  return globalConfigModel.findOne({
+    where: {
+      key: dbConstants.SGEncryptionKey,
```

Comment:
why two separate db select queries? can't you do this in one db select, find where (key = x OR key = y)?

Reply:
A single query will return an array of 2 elements instead. So, we have to go through the array, do some string comparison before returning the object back vs sending 2 queries on an indexed field. Do you think we should go with the array approach?

Reply:
i suspect manipulating an array of two elements is cheaper than two roundtrip db queries, but we're talking about such small numbers it doesn't really matter